lua/haskell-snippets/expressions.lua

```diff
@@ -49,6 +50,23 @@ expressions.if_expr_multiline = s({
 })
 table.insert(expressions.all, expressions.if_expr_multiline)
 
+expressions.if_expr_multiway = s({
+  trig = 'ifmw',
+  dscr = 'If expression (multi-way)',
+}, {
+  text('if | '),
+  insert(1),
+  text(' -> '),
+  insert(2),
+  text { '', '   | ' },
+  insert(3),
+  text(' -> '),
+  insert(4),
+  text { '', '   | otherwise -> ' },
+  insert(5),
```

Comment:
```suggestion
  text('if | '),
  insert(1),
  text(' -> '),
  insert(2),
  text { '', util.indent_newline_text('| ') },
  insert(3),
  text(' -> '),
  insert(4),
  text { '', util.indent_newline_text('| otherwise -> ')},
  insert(5),
```

Reply:
I have applied util.indent_newline_text to my commit, but I haven't used this particular suggestion, because it has a few bugs, e.g., it throws this error:

```
Error executing vim.schedule lua callback: ....local/share/nvim/lazy/LuaSnip/lua/luasnip/util/util.lua:55: attempt to concatenate a function value                                                                                                                                                          
stack traceback:                                                                                                                                                                                                                                                                                            
        ....local/share/nvim/lazy/LuaSnip/lua/luasnip/util/util.lua:55: in function 'indent'                                                                                                                                                                                                                
        ...local/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/node.lua:265: in function 'indent'                                                                                                                                                                                                               
        ...[al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:1008](http://al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:1008): in function 'indent'                                                                                                                                                                                                              
        ...[al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:895](http://al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:895): in function 'fake_expand'                                                                                                                                                                                                          
        ...[al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:959](http://al/share/nvim/lazy/LuaSnip/lua/luasnip/nodes/snippet.lua:959): in function 'get_docstring'                                                                                                                                                                                                        
        ...[cal/share/nvim/lazy/cmp_luasnip/lua/cmp_luasnip/init.lua:37](http://cal/share/nvim/lazy/cmp_luasnip/lua/cmp_luasnip/init.lua:37): in function 'get_documentation'                                                                                                                                                                                                     
        ...[cal/share/nvim/lazy/cmp_luasnip/lua/cmp_luasnip/init.lua:121](http://cal/share/nvim/lazy/cmp_luasnip/lua/cmp_luasnip/init.lua:121): in function 'resolve'                                                                                                                                                                                                              
        ...esiek/.local/share/nvim/lazy/nvim-cmp/lua/cmp/source.lua:392: in function 'resolve'                                                                                                                                                                                                              
        ...zesiek/.local/share/nvim/lazy/nvim-cmp/lua/cmp/entry.lua:495: in function 'resolve'                                                                                                                                                                                                              
        ...rzesiek/.local/share/nvim/lazy/nvim-cmp/lua/cmp/view.lua:266: in function 'fn'                                                                                                                                                                                                                   
        .../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/async.lua:71: in function <.../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/async.lua:69>
  ```